### PR TITLE
feat: search

### PR DESCRIPTION
**Pull request by Fupete**
Partially fix: #408 

Now: 
- [ ] config and test the search engine
  - [x] first config
  - [ ] user testing
  - [ ] improvements
- [ ] main home page search field
  - [x] basic draft
  - [x] show content type (from id/gatsby)
  - [x] activate suggested search buttons
  - [x] a11y testing 
  - [ ] user testing
  - [ ] improvements
  - [ ] switch to use a full list and not just list-item component (dropdown?)
- [ ] result list page
  - [x] basic draft
  - [ ] pagination 
  - [ ] sort (score, alphabetical, date of update)
- [ ] header search 
  - [x] button to "ricerca" page
  - [ ] basic draft field flow
  - [ ] user testing
  - [ ] improvements 

Future: 
- [ ] advanced result list page with contextual kangaroo's data dropdown-select filters

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -101,7 +101,152 @@ module.exports = {
...
-        engineOptions: "speed",
+        engineOptions: {
+          profile: "score",
+          charset: "latin:advanced",
+          // encode: "extra",
+          tokenize: "reverse",
+          threshold: 11,
+          resolution: 22,
+          depth: 2,
+          filter: [
```
Comment (bfabio) [inline gatsby-config.js:112]: I'd expect the library to do this for us if we say we're dealing with Italian. Same thing for the stemmer.

If flexsearch doesn't support this out of the box we should consider upstreaming the support for Italian.

Comment (Fupete) [reply gatsby-config.js:112]: Hi @bfabio, the library support `en` and `de` native. We can add our `it` filters by file, or like this, during initialization setup. IMHO, using our Gatsby plugin local-search to integrate flexsearch, and counting that we are using an old and deprecated version 6, this second way, during build time, is the only way right now.

**Comment by Fupete** [comment]
Post-it: all for today. Tomorrow I will work on the results page pagination (example settings: `limit: 20, page: true`), this is the [reference](https://github.com/nextapps-de/flexsearch/tree/0.6.32#pagination). I think we need also a pagination component from Bootstrap Italia https://italia.github.io/bootstrap-italia/docs/componenti/paginazione/ to trigger each 'reload/page switch'... is it right @bfabio?

**Comment by Fupete** [comment]
@bfabio I think we can review & merge this PR, ready for usability testing next week. 
I would not close the related issue, for now, to continue working on what is missing and archives.

Let's talk about it tomorrow at our synchronization meeting, that is, right after the a11y review.

File: src/components/search-main/search-main.js [hunk 2]
```diff
@@ -40,16 +68,35 @@ const SearchMain =({
...
+  const searchOptions = {
+    limit: `${howMany ? howMany : ''}`,
+    suggest: `${useSuggestionEngine ? true : false}`, // XXX < activate to "fill" the spaces with suggestions, usefull for multiple words query
+  }
+
+  const results = useFlexSearch(searchTerm, index, store, searchOptions) // XXX < search run
 
   let styles = 'search-main'
 	+ `${background ? ' bg-'+background : ''}`
+
```
Comment (Fupete) [inline src/components/search-main/search-main.js:89]: @Fupete we can remove this

File: gatsby-config.js [hunk 3]
```diff
@@ -107,7 +107,153 @@ module.exports = {
...
+            "ha",
+            "hai",
+            "hanno",
+            "ho",
+            "il",
+            "indietro", 
+            "invece",
+            "io",
+            "la",
+            // "lavoro",
```
Comment (bfabio) [inline gatsby-config.js:174]:
```suggestion
```

File: gatsby-config.js [hunk 4]
```diff
@@ -107,7 +107,153 @@ module.exports = {
...
+            "anche",
+            "ancora",
+            "avere",
+            "aveva",
+            "avevano",
+            "ben",
+            "buono",
+            "che",
+            "chi",
+            // "cinque",
```
Comment (bfabio) [inline gatsby-config.js:140]:
```suggestion
```

File: src/components/search-main/search-main.js [hunk 5]
```diff
@@ -17,19 +23,42 @@ const SearchMain =({
...
-  useEffect(() => {
-    setSearchEnabled(window.localStorage.getItem("search-enabled") ? 1 : 0)
-  }, [])
+  const [input, setInput] = useState(() => {
+    if (location)
+      if (location.state) 
+        return location.state.searchTerm
+      else 
+        return null
+  })
```
Comment (bfabio) [inline src/components/search-main/search-main.js:33]:
```suggestion
  const [input, setInput] = useState(() => location?.state?.searchTerm)
```

File: src/components/search-main/search-main.js [hunk 6]
```diff
@@ -40,65 +69,118 @@ const SearchMain =({
...
+  const search = (input) => {
     setSearchTerm(input)
-    console.log(index, store)
+    setFormSubmitted(true)
+    setStoredInput(input)
+  }
+
+  const searchOptions = {
+    limit: `${howMany ? howMany : ''}`,
+    suggest: useSuggestionEngine, // XXX < activate to "fill" the spaces with suggestions, usefull for multiple words query
```
Comment (bfabio) [inline src/components/search-main/search-main.js:87]:
```suggestion
    // if true "fill" the spaces with suggestions, useful for multiple words query
    suggest: useSuggestionEngine, // XXX < 
```

File: src/components/search-main/search-main.js (hunk 5 above)
Comment (bfabio) [reply src/components/search-main/search-main.js:33]: (same thing for the others `useState()` below as well)

**Pull request merged**